packages/typography/src/Link.spec.tsx

```diff
@@ -56,7 +56,7 @@ describe('packages/typography', () => {
         });
         const anchor = screen.getByText('Link');
         const icon = screen.queryByLabelText('Arrow Right Icon');
-        expect(icon).not.toBeVisible();
+        // expect(icon).not.toBeVisible();
```

Comment:
So this is a faulty test in general. We're testing for a lack of visibility by default and then not checking that it is actually visible when we hover - only that the Icon is in the document (which doesn't prove anything). When I try and update the test to be icon.toBeVisible() that fails too. I can inspect the css styles in the browser-rendered version of the test suite and I see that the styles are correctly applied -- the only thing that I can suspect that's preventing React Testing library from evaluating the Icon as visible is that it's not a straightforward css rule that transitions the opacity, and maybe that's causing the issue. Have been struggling with this for a while, and I'm not too sure what to do here.

Reply:
I'm leaning towards removing the test entirely and filing a ticket to create a better test here. I'm almost positive that this Icon updates did not break the test, but rather exposed that the test never worked as expected in the first place. What do you think @lazytype

Reply:
> the only thing that I can suspect that's preventing React Testing library from evaluating the Icon as visible is that it's not a straightforward css rule that transitions the opacity

for transitions I've had some success by waiting for them to finish, e.g. `await waitFor(() => expect(icon).not.toBeVisible());`  (`waitForElementToBeRemoved` if the element is no longer in the document at all). if this approach still doesn't work then I'm ok with creating a ticket and linking it in the file

Reply:
Still doesn't work, even though I'm seeing this rule applied: 

`[data-leafygreen-ui="anchor-container"]:hover .leafygreen-ui-vo4x0d{opacity:1;-webkit-transform:translate3d(3px,0,0);-ms-transform:translate3d(3px,0,0);transform:translate3d(3px,0,0);}`

Reply:
Rming this test: 

```
test('and the "arrowAppearance" prop is set to "hover"', async () => {
        renderLink({
          href: 'http://localhost:9001',
          arrowAppearance: 'hover',
        });
        const anchor = screen.getByText('Link').parentNode;
        const icon = screen.queryByLabelText('Arrow Right Icon');
        expect(icon).not.toBeVisible();

        fireEvent.mouseEnter(anchor);
        await waitFor(() => expect(icon).toBeVisible())
      });
```

Reply:
https://jira.mongodb.org/browse/PD-1090